library_generation/utils/commit_message_formatter.py

```diff
@@ -47,19 +47,24 @@ def format_commit_message(commits: Dict[Commit, str], is_monorepo: bool) -> List
                 messages.append(formatted_message)
             else:
                 messages.append(message_line)
-        all_commits.extend(wrap_nested_commit(messages))
+        all_commits.extend(wrap_nested_commit(commit, messages))
     return all_commits
 
 
-def wrap_nested_commit(messages: List[str]) -> List[str]:
+def wrap_nested_commit(commit: Commit, messages: List[str]) -> List[str]:
     """
     Wrap message between `BEGIN_NESTED_COMMIT` and `BEGIN_NESTED_COMMIT`.
 
+    :param commit: a GitHub commit.
     :param messages: a (multi-line) commit message, one line per item.
     :return: wrapped messages.
     """
+    short_sha = commit.hexsha[:7]
     result = ["BEGIN_NESTED_COMMIT"]
     result.extend(messages)
+    result.append(
+        f"Source Link: [googleapis/googleapis@{short_sha}](https://github.com/googleapis/googleapis/commit/{commit.hexsha})"
```

Comment:
Optional nit:

FWIW Github will automatically format the link if you just write the link in plain text. That way we make the logic simple at the cost of relying on the service (is it stable?).

E.G. pasting `https://github.com/googleapis/googleapis/commit/19af522bf3ab49b7d4ce284a568b2e8bb28aeae3` results in https://github.com/googleapis/googleapis/commit/19af522bf3ab49b7d4ce284a568b2e8bb28aeae3

Reply:
Thanks for the suggestion, though I don't want to depends on github.